### PR TITLE
Adding reusable workflow to repo

### PR DESCRIPTION
**Pull request by nestor-sk**
This adds the reusable workflow and an example of how to use it.

### CODE REVIEW

File: .github/workflows/on-pr-example.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+permissions: 
+  contents: read
+  checks: write
+  pull-requests: write
+
+concurrency: on-pr-example-concurrency
+
+jobs:
+  gql-lint:
+    uses: sketch-hq/gql-lint-integrations/.github/workflows/gql_new_deprecated_fields.yml@main
```
Comment (omh) [inline .github/workflows/on-pr-example.yml:14]: This URL is old, right?

Comment (nestor-sk) [reply .github/workflows/on-pr-example.yml:14]: 🕵️  I was changing the others but forgot about this one :) Fix incoming

Comment (nestor-sk) [reply .github/workflows/on-pr-example.yml]: https://github.com/sketch-hq/gql-lint/pull/34/commits/04069f5883d645a29b7dea1c4a68906b3af6c85f

**Review by omh — APPROVED** [review]
:shipit: ✨

**Pull request merged**